### PR TITLE
Separated to hooks, models and comps and added Mutation for favorite

### PR DESCRIPTION
**Pull request by alonw0**
Hi again!
I separated the code to hooks, components and models like we learned + Added the Mutation.

Have a good week!

### CODE REVIEW

File: src/components/PopularTags.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import { useQuery } from 'react-query'
+
+export default function PopularTags({ onClick }) {
+  const { data, isFetching } = useQuery(['/tags', {}], {
+    placeholderData: {
+      tags: [],
+    },
+  })
+  console.log('DATA:', data)
```
Comment (romansndlr) [inline src/components/PopularTags.jsx:10]: Remove please

File: src/components/PopularTags.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import { useQuery } from 'react-query'
+
+export default function PopularTags({ onClick }) {
+  const { data, isFetching } = useQuery(['/tags', {}], {
```
Comment (romansndlr) [inline src/components/PopularTags.jsx:5]:
```suggestion
  const { data, isFetching } = useQuery('/tags', {
```

File: src/main.jsx [hunk 3]
```diff
@@ -2,10 +2,25 @@ import React from 'react'
...
 import { ReactQueryDevtools } from 'react-query/devtools'
-
+import axios from 'axios'
 import App from './App'
 
-const queryClient = new QueryClient()
+axios.defaults.baseURL = 'https://conduit.productionready.io/api'
+
+const defaultQueryFn = async ({ queryKey }) => {
+  console.log(queryKey)
```
Comment (romansndlr) [inline src/main.jsx:11]: Remove

File: src/main.jsx [hunk 4]
```diff
@@ -2,10 +2,25 @@ import React from 'react'
...
+  const { data } = await axios.get(queryKey[0], { params: queryKey[1] })
+  console.log(data)
```
Comment (romansndlr) [inline src/main.jsx:13]: Remove

File: src/pages/Home.jsx [hunk 5]
```diff
@@ -3,42 +3,17 @@ import axios from 'axios'
 import classNames from 'classnames'
 import React from 'react'
 import { useQuery } from 'react-query'
+import { Article, PopularTags } from '../components'
+import useArticle from '../hooks/useArticle'
```
Comment (romansndlr) [inline src/pages/Home.jsx]:
```suggestion
import { useArticle } from '../hooks'
```

**Review by romansndlr — APPROVED** [review]
Looks great! Awesome work! Please be sure to go over my comments.

File: src/components/PopularTags.jsx (hunk 2 above)
Comment (alonw0) [reply src/components/PopularTags.jsx:5]: Done

**Pull request merged**